stats.go

```diff
@@ -112,6 +115,51 @@ func startPrometheusStats(l *logrus.Logger, i time.Duration, c *config.C, buildV
 	pr.MustRegister(g)
 	g.Set(1)
 
+	// Export the start time of the current binary execution
+	start_time := prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		Name:        "start_time",
+		Help:        "Time the nebula client started, in unixtime",
+		ConstLabels: prometheus.Labels{},
+	})
+	pr.MustRegister(start_time)
+	start_time.Set(float64(time.Now().Unix()))
+
+	// Export our certificate information as labels on a static gauge
+	for _, ca := range caCert.CAs {
+
+		ca_info := prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "ca_cert_expiry",
+			Help:      "Returns CA cert time expiry in unixtime",
+			ConstLabels: prometheus.Labels{
+				// TODO : fill these in with actual data, not hardcoded fields
+				"name":       ca.Details.Name,
+				"public_key": hex.EncodeToString(ca.Details.PublicKey),
+			},
+		})
+
+		pr.MustRegister(ca_info)
+		ca_info.Set(float64(ca.Details.NotAfter.Unix()))
```

Comment:
I think it would be easier to write alerts if this tracked the remaining seconds until expiry: `remaining time < 14 days`.

It could also be interesting to report how long the cert was issued for. An example would be to warn at the half life: `remaining time / valid lifetime <= 0.5`

Reply:
I considered that, but this sort of thing seems to be a bit of a prometheus anti-pattern. The generally accepted practice around prometheus metrics is when something is static, it should be reported as static and write the alert around that. I can update it to your method if you insist, but it would be breaking the typical way people use prometheus.

Thoughts?

Reply:
Examples of this are `node_boot_time_seconds` in the prometheus node exporter, and `ssl_earliest_cert_expiry` in the blackbox exporter, both of which are first-party prometheus exporters maintained by the core maintainers of prometheus.

Reply:
There are a few complicating factors. Certificates can be issued with different lifespans. You may want some nodes with certs that live for hours while others live for months or years. We would at least need to add `NotBefore` so that we can determine the lifespan.

Exporting only the static information is likely the most widely applicable but also the most frustrating to implement alerts for.

Maybe @wadey has some thoughts

Reply:
Adding `NotBefore` makes sense to me. I can do that.

However, on the alerting point, I'd argue it would be far more user-hostile to go against prometheus typical usage patterns. Anyone already using prometheus for other things (like the Blackbox exporter to monitor) web certificates will already be deeply familiar with this pattern. The key thing is that anyone using prometheus is unlikely to be using it *only* to monitor nebula, it's a broader ecosystem.

Again, I'd be happy to adopt your suggestion if you insist, but I think it will be user-hostile for prometheus users.

Reply:
No I think we can get by with `NotBefore` and `NotAfter`. 

Alerting based on a percentage of consumed lifespan could be less burdensome than `(nebula_client_client_cert_not_after - time()) / (nebula_client_client_cert_not_after - nebula_client_client_cert_not_before) <= 0.5`, so maybe we want an to expose both styles (possibly optionally).

A sampling of the exporters that live out there show a mix of approaches
- Duration values https://github.com/joe-elliott/cert-exporter (grafana advises the use of this one https://grafana.com/blog/2020/03/04/pro-tip-how-to-monitor-client-certificate-expirations-with-prometheus/)
- Static values https://github.com/ribbybibby/ssl_exporter
- Duration values https://github.com/amimof/node-cert-exporter
- Static values https://github.com/enix/x509-certificate-exporter

Reply:
For what it's worth, the alert query string you noted above seems pretty reasonable to me, but I've also been using prometheus for a while. I have many alert queries even more complex than that, and most other non-trivial prometheus users would too.

I don't deny that some smaller exporters (all the ones linked are under 300 stars) might do different things, but it does still seem worth trying to follow the standard created and recommended by the prometheus maintainers where possible, unless there's a really strong reason not to do so.

Is there a use-case I'm missing where a substantial amount of people will be using prometheus to alert only on nebula and nothing else?

Reply:
Alternatively, what if I update the PR to provide a documentation section on suggested alerts? This is something I would like to do anyways, I could roll it in with this one, and get rid of the need of anyone to figure out the alert string. I'd also like to get the alert strings put here too (https://awesome-prometheus-alerts.grep.to) if they accept it.